tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -96,65 +100,68 @@ def test_non_ocs_taint_and_tolerations(self):
+            resource_name=constants.ROOK_OPERATOR_CONFIGMAP,
         )
-        for sub in sub_list:
-            sub_obj = ocp.OCP(
-                resource_name=sub,
-                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
-                kind=constants.SUBSCRIPTION,
-            )
-            sub_obj.patch(params=param, format_type="merge")
+        toleration = (
+            '\n- key: xyz\n  operator: Equal\n  value: "true"\n  effect: NoSchedule'
+        )
+        toleration = toleration.replace('"', '\\"').replace("\n", "\\n")
+
+        params = (
+            f'{{"data": {{"CSI_PLUGIN_TOLERATIONS": "{toleration}", '
+            f'"CSI_PROVISIONER_TOLERATIONS": "{toleration}"}}}}'
+        )
+
+        configmap_obj.patch(params=params, format_type="merge")
+        logger.info(f"Successfully added toleration to {configmap_obj.kind}")
 
         # Add tolerations to the ocsinitializations.ocs.openshift.io
         param = (
             '{"spec":  {"tolerations": '
             '[{"effect": "NoSchedule", "key": "xyz", "operator": "Equal", '
             '"value": "true"}]}}'
         )
-
         ocsini_obj = ocp.OCP(
             resource_name=constants.OCSINIT,
             namespace=defaults.ROOK_CLUSTER_NAMESPACE,
             kind=constants.OCSINITIALIZATION,
         )
         ocsini_obj.patch(params=param, format_type="merge")
+        logger.info(f"Successfully added toleration to {ocsini_obj.kind}")
 
-        # Add tolerations to the configmap rook-ceph-operator-config
-        configmap_obj = ocp.OCP(
-            kind=constants.CONFIGMAP,
-            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
-            resource_name=constants.ROOK_OPERATOR_CONFIGMAP,
-        )
-        toleration = configmap_obj.get().get("data").get("CSI_PLUGIN_TOLERATIONS")
-        toleration += (
-            '\n- key: xyz\n  operator: Equal\n  value: "true"\n  effect: NoSchedule'
-        )
-        toleration = toleration.replace('"', '\\"').replace("\n", "\\n")
-        param_cmd = (
-            f'[{{"op": "replace", "path": "/data/CSI_PLUGIN_TOLERATIONS", "value": "{toleration}" }}, '
-            f'{{"op": "replace", "path": "/data/CSI_PROVISIONER_TOLERATIONS", "value": "{toleration}" }}]'
+        # Add tolerations to the subscription
+        sub_list = ocp.get_all_resource_names_of_a_kind(kind=constants.SUBSCRIPTION)
+        param = (
+            '{"spec": {"config":  {"tolerations": '
+            '[{"effect": "NoSchedule", "key": "xyz", "operator": "Equal", '
+            '"value": "true"}]}}}'
         )
-        configmap_obj.patch(params=param_cmd, format_type="json")
+        for sub in sub_list:
+            sub_obj = ocp.OCP(
+                resource_name=sub,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                kind=constants.SUBSCRIPTION,
+            )
+            sub_obj.patch(params=param, format_type="merge")
+            logger.info(f"Successfully added toleration to {sub}")
```

Comment:
I would suggest having an order of editing. First storagecluster, subscription, configmap and then toolbox